eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Mempool.hs

```diff
@@ -100,24 +102,42 @@ mempoolTransition ::
 mempoolTransition = do
   TRC trc@(_ledgerEnv, ledgerState, tx) <-
     judgmentContext
+
   -- This rule only gets invoked on transactions within the mempool.
   -- Add checks here that sanitize undesired transactions.
+
+  let whenFailureFree' action = ifFailureFree action (pure ledgerState)
+
+  -- Detect whether the transaction is probably a duplicate
   let
-    authorizedElectedHotCreds = authorizedElectedHotCommitteeCredentials ledgerState
-    collectUnelectedCommitteeVotes !unelectedHotCreds voter _ =
-      case voter of
-        CommitteeVoter hotCred
-          | hotCred `Set.notMember` authorizedElectedHotCreds ->
-              Set.insert hotCred unelectedHotCreds
-        _ -> unelectedHotCreds
-    unelectedCommitteeVoters =
-      Map.foldlWithKey' collectUnelectedCommitteeVotes Set.empty $
-        unVotingProcedures (tx ^. bodyTxL . votingProceduresTxBodyL)
-    addPrefix =
-      ("Unelected committee members are not allowed to cast votes: " <>)
-  failOnNonEmpty unelectedCommitteeVoters $
-    ConwayMempoolFailure . addPrefix . T.pack . show . NE.toList
-  trans @(EraRule "LEDGER" era) $ TRC trc
+    inputs = tx ^. bodyTxL . inputsTxBodyL
+    utxo = ledgerState ^. utxoG . to unUTxO
+    unspent = (`Map.member` utxo)
+    msg = "All inputs are spent. Transaction has probably already been included"
+  any unspent inputs ?! ConwayMempoolFailure msg
+
+  -- Skip all other checks if the transaction is probably a duplicate
+  whenFailureFree' $
+    do
```

Comment:
Please try to avoid this style. I am trying to promote somewhat consistent Haskell style in the codebase, this would be one that is inconsistent.
```suggestion
  whenFailureFreeDefault ledgerState $ do
```
Also this has my other suggestion applied.

Reply:
I did it that way because I thought `fourmolu` would change it, but apparently it's OK with yours. Thanks for catching this. I too prefer it your way.